src/libraries/System.Net.Connections/tests/System.Net.Connections.Tests/Sockets/SocketsConnectionFactoryTests.cs

```diff
@@ -0,0 +1,261 @@
+        {
+            Assert.True(connection.ConnectionProperties.TryGet(out Socket socket));
+            Assert.True(socket.Connected);
+            if (addressFamily != null) Assert.Equal(addressFamily, socket.AddressFamily);
+            if (socketType != null) Assert.Equal(socketType, socket.SocketType);
+            if (protocolType != null) Assert.Equal(protocolType, socket.ProtocolType);
+            return socket;
+        }
+
+        [Theory]
+        [MemberData(nameof(GetConnectData))]
+        public async Task Constructor3_ConnectAsync_Success_PropagatesConstructorArgumentsToSocket(EndPoint endPoint, SocketType socketType, ProtocolType protocolType)
+        {
+            endPoint = RecreateUdsEndpoint(endPoint);
+            using var server = SocketTestServer.SocketTestServerFactory(SocketImplementationType.Async, endPoint, protocolType);
+            using SocketsConnectionFactory factory = new SocketsConnectionFactory(endPoint.AddressFamily, socketType, protocolType);
+            using Connection connection = await factory.ConnectAsync(server.EndPoint);
+
+            ValidateSocket(connection, socketType, protocolType, endPoint.AddressFamily);
+        }
+
+        [Fact]
+        public async Task Constructor2_ConnectAsync_Success_CreatesIPv6DualModeSocket()
+        {
+            using var server = SocketTestServer.SocketTestServerFactory(SocketImplementationType.Async, IPAddress.IPv6Loopback);
+            using SocketsConnectionFactory factory = new SocketsConnectionFactory(SocketType.Stream, ProtocolType.Tcp);
+            using Connection connection = await factory.ConnectAsync(server.EndPoint);
+
+            Socket socket = ValidateSocket(connection, SocketType.Stream, ProtocolType.Tcp, AddressFamily.InterNetworkV6);
+            Assert.True(socket.DualMode);
+        }
+
+        [Fact]
+        public async Task ConnectAsync_Success_SocketNoDelayIsTrue()
+        {
+            using var server = SocketTestServer.SocketTestServerFactory(SocketImplementationType.Async, IPAddress.Loopback);
+            using SocketsConnectionFactory factory = new SocketsConnectionFactory(SocketType.Stream, ProtocolType.Tcp);
+            using Connection connection = await factory.ConnectAsync(server.EndPoint);
+
+            connection.ConnectionProperties.TryGet(out Socket socket);
+            Assert.True(socket.NoDelay);
+        }
+
+        [Fact]
+        public void ConnectAsync_NullEndpoint_ThrowsArgumentNullException()
+        {
+            using SocketsConnectionFactory factory = new SocketsConnectionFactory(SocketType.Stream, ProtocolType.Tcp);
+            Assert.ThrowsAsync<ArgumentNullException>(() => factory.ConnectAsync(null).AsTask());
+        }
+
+        // TODO: On OSX and Windows7 connection failures seem to fail with unexpected SocketErrors that are mapped to NetworkError.Unknown. This needs an investigation.
+        // Related: https://github.com/dotnet/runtime/pull/40565
```

Comment:
All of these tests are failing with unexpected SocketErrors? That seems really bad. Do we understand what's happening here?

Reply:
Maybe "unexpected" is a bad word, and it's a known behavior on those platforms. *Seems* related that some of the `Connect`  tests are disabled on OSX.

We should remember to open an issue about this if there's none yet.